controllers/ibmblockcsi_controller_test.go

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	csiv1 "github.com/IBM/ibm-block-csi-operator/api/v1"
+	"github.com/IBM/ibm-block-csi-operator/pkg/config"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+  
+  var _ = Describe("Controller", func() {
+  
+	  const timeout = time.Second * 30
+	  const interval = time.Second * 1
+	  var ibc *csiv1.IBMBlockCSI
+	  var ns = "default"
+	  var ibcName = "test-ibc"
+	  var apiVersion = "csi.ibm.com/v1"
+	  var kind = "IBMBlockCSI"
+	  containersImages := map[string]string{
+		"ibm-block-csi-controller":  "fake-controller-repo:fake-controller-tag",
+		"ibm-block-csi-node": 		 "fake-node-repo:fake-node-tag",
+		"csi-provisioner": 			 "fake-provisioner-repo:fake-provisioner-tag",
+		"csi-attacher": 			 "fake-attacher-repo:fake-attacher-tag",
+		"csi-snapshotter": 			 "fake-snapshotter-repo:fake-snapshotter-tag",
+		"csi-resizer": 				 "fake-resizer-repo:fake-resizer-tag",
+		"csi-addons-replicator": 	 "fake-replicator-repo:fake-replicator-tag",
+		"livenessprobe": 			 "fake-livenessprobe-repo:fake-livenessprobe-tag",
+		"csi-node-driver-registrar": "fake-registrar-repo:fake-registrar-tag",
```

Comment:
could we just take the containers details dynamically from the cr? I don't think we want to maintain all these strings

Reply:
done